deepinv/sampling/diffusion.py

```diff
@@ -175,6 +175,171 @@ def forward(self, y, physics: deepinv.physics.DecomposablePhysics, seed=None):
         return x
 
 
+class DiffPIR(nn.Module):
+    r"""
+    Diffusion PnP Image Restoration (DiffPIR).
+
+    This class implements the Diffusion PnP image restoration algorithm (DiffPIR) described in https://arxiv.org/abs/2305.08995.
+
+    The DiffPIR algorithm is an HQS PnP algorithm, where the denoiser is a conditional diffusion denoiser,
+    combined with a diffusion process.
+
+    :param torch.nn.Module model: a conditional noise estimation model
+    :param float sigma: the noise level of the data
```

Comment:
That's a good question we should consider. Currently I would say no because the current normal thing to do with diffusion is to use the time embedding (as opposed to sigma) so I am afraid that by choosing sigma instead of t we make the algo less clear to the diffusion community. But I think that in the future we should maybe try to make it more consistent among methods...

Reply:
Good idea, but I would not necessarily put physics.noise_model.sigma because it might not be the same value, I've added as an optional input in the forward.